src/Voron/Platform/Posix/PosixMemoryMapPager.cs

```diff
@@ -20,7 +20,7 @@ public PosixMemoryMapPager(int pageSize,string file, long? initialFileSize = nul
 
             PosixHelper.EnsurePathExists(_file);
 
-            _fd = Syscall.open(file, OpenFlags.O_RDWR | OpenFlags.O_CREAT | OpenFlags.O_SYNC,
+            _fd = Syscall.open(file, OpenFlags.O_RDWR | OpenFlags.O_CREAT,
```

Comment:
I'm feeling that this is bad. We have both PosixMemoryMapPager and PosixTempMemoryMapPager, why the difference?

It looks like we are using the temp just for pure memory stuff. But it looks like we can use it for things like scratch buffers in general.

Regardless, we shouldn't have O_SYNC here, btw